### PR TITLE
Step rewrite (+ misc stuff)

### PR DESCRIPTION
**Pull request by Doogie13**
Step now doesnt rubberband almost _at all_ on strict with timer and height 1 and works better than most

Misc stuff: 
- Speed now has strict jump height for cc and 2b2t(.dev)
- HUD speed element now multiplies by timer speed making it more accurate

### CODE REVIEW

**Comment by Doogie13** [comment]
1 block step occasionally rubberbands when NCP has a fit but its better than whatever we had before for step

**Comment by Doogie13** [comment]
wait dont merge this yet... just realised the HUD element doesnt calculate timer properly kek

**Comment by Doogie13** [comment]
fixed

File: src/main/java/cope/cosmos/asm/mixins/MixinEntity.java [hunk 1]
```diff
@@ -2,16 +2,224 @@
...
+    public void move(MoverType type, double tx, double ty, double tz, CallbackInfo ci) {
+
+        if (Minecraft.getMinecraft().getCurrentServerData() == null)
+            return;
+
+        double x = tx;
+        double y = ty;
+        double z = tz;
+
+        if (ci.isCancelled())
```
Comment (yagel15637) [inline src/main/java/cope/cosmos/asm/mixins/MixinEntity.java:35]: what the fuck

File: src/main/java/cope/cosmos/asm/mixins/MixinEntity.java [hunk 2]
```diff
@@ -2,16 +2,224 @@
...
+                }
+
+                bb = (bb.offset(0.0D, y, 0.0D));
+
+                if (!(d14 * d14 + d7 * d7 >= x * x + z * z)) {
+
+                    StepEvent event = new StepEvent(bb,stepHeight);
+                    Cosmos.EVENT_BUS.post(event);
+
+                    Minecraft.getMinecraft().player.stepHeight = event.getHeight();
```
Comment (yagel15637) [inline src/main/java/cope/cosmos/asm/mixins/MixinEntity.java:215]: what is this pasted mess

File: src/main/java/cope/cosmos/client/events/StepEvent.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package cope.cosmos.client.events;
+
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+public class StepEvent extends Event {
```
Comment (yagel15637) [inline src/main/java/cope/cosmos/client/events/StepEvent.java:6]: why

File: src/main/java/cope/cosmos/client/features/modules/movement/Step.java [hunk 4]
```diff
@@ -42,57 +35,55 @@ public void onDisable() {
...
+            return;
+
+        if (!mc.player.onGround) {
+            event.setHeight(.6f);
+            return;
         }
+
+        double step = event.getBB().minY - mc.player.posY;
+
+        if (step == 1)
```
Comment (yagel15637) [inline src/main/java/cope/cosmos/client/features/modules/movement/Step.java:57]: oh lord.

Comment (Doogie13) [reply src/main/java/cope/cosmos/client/features/modules/movement/Step.java:57]: if it works, it works

Comment (yagel15637) [reply src/main/java/cope/cosmos/client/features/modules/movement/Step.java:57]: bad mindset that wont get you anywhere 👎

**Pull request merged**